src/main/java/bio/terra/common/sam/SamRetry.java

```diff
@@ -27,11 +26,13 @@
   private static final Duration OPERATION_TIMEOUT = Duration.ofSeconds(300);
   private final Instant operationTimeout;
 
+  // Sam calls which timeout will throw ApiExceptions wrapping SocketTimeoutExceptions and will have
+  // an errorCode 0. This isn't a real HTTP status code, but we can check for it anyway.
+  private static final int TIMEOUT_STATUS_CODE = 0;
```

Comment:
Is there anything else in the ApiException we can use to check if it's a time out exception? It might be better to be more explicit rather than retry anything that doesn't have a status code and defaults to 0.
Not sure how you can figure this out other than wait for Sam to socket timeout and look at the exception...

Reply:
Good point, we can also look at the `cause` field which will have a `SocketTimeoutException` for timeouts. 
(and yep, that was how I found the 0 status code)